freqtrade/freqai/RL/TensorboardCallback.py

```diff
@@ -44,7 +45,10 @@ def _on_training_start(self) -> None:
     def _on_step(self) -> bool:
 
         local_info = self.locals["infos"][0]
-        tensorboard_metrics = self.training_env.get_attr("tensorboard_metrics")[0]
+        if isinstance(self.training_env, SubprocVecEnv):
+            tensorboard_metrics = self.training_env.get_attr("tensorboard_metrics")[0]
+        else:
+            tensorboard_metrics = self.training_env.tensorboard_metrics
```

Comment:
@robcaulk according to debugging one of the tests, this is of type  `VecMonitor` - not of type BaseEnvironment.
I suspect the correct type for training_env would therefore be `Union[SubprocVecEnv, VecMonitor]` ...

Reply:
It depends, if they use `_multiproc` then it will be a VecMonitor/SubProcVecEnv. If they are using the regular `ReinforcementLearner` then it will be of type `BaseEnvironment`. Yes, this is the downside of trying to support both methods, fun typing.